### PR TITLE
DB schema and matching types

### PR DESCRIPTION
**Pull request by AmineAfia**

### CODE REVIEW

**Comment by AmineAfia** [comment]
* **#30** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/30?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @AmineAfia and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: internal/worker/serializer.go [hunk 1]
```diff
@@ -46,15 +46,19 @@ func serializeBlock(rpc common.RPC, block *types.Block) common.Block {
...
+		LogsBloom:        block.Bloom().Big().String(),
+		Size:             uint64(block.Size()),
 		ExtraData:        string(block.Extra()),
+		Difficulty:       block.Difficulty(),
 		GasLimit:         big.NewInt(int64(block.GasLimit())),
 		GasUsed:          big.NewInt(int64(block.GasUsed())),
 		TransactionCount: uint64(len(block.Transactions())),
-		BaseFeePerGas:    block.BaseFee(),
+		BaseFeePerGas:    block.BaseFee().Uint64(),
 		WithdrawalsRoot: func() string {
```
Comment (iuwqyir) [inline internal/worker/serializer.go:62]: this should be `block.Header().WithdrawalsHash.Hex()`

File: internal/worker/serializer.go [hunk 2]
```diff
@@ -46,15 +46,19 @@ func serializeBlock(rpc common.RPC, block *types.Block) common.Block {
...
 		Sha3Uncles:       block.UncleHash().Hex(),
-		LogsBloom:        block.Bloom().Big().String(),
+		MixHash:          block.Header().MixDigest.Hex(),
+		Miner:            block.Header().Coinbase.Hex(),
+		StateRoot:        block.Header().Root.Hex(),
+		TransactionsRoot: block.Header().TxHash.Hex(),
 		ReceiptsRoot:     block.ReceiptHash().Hex(),
-		Difficulty:       block.Difficulty(),
-		Size:             float64(block.Size()),
+		LogsBloom:        block.Bloom().Big().String(),
```
Comment (iuwqyir) [inline internal/worker/serializer.go:54]:
```suggestion
		LogsBloom:        block.Bloom().Big().Text(16),
```

**Review by iuwqyir — COMMENTED** [review]
Looks good, just need to change 2 block serializations

**Comment by iuwqyir** [comment]
> Looks good, just need to change 2 block serializations

changed it

**Pull request merged**